### PR TITLE
Modified how the next clone is selected when switching and added a me…

### PR DESCRIPTION
**Pull request by EvidentlyCube**
…chanism that ensures the clones are always in the same movement order regardless of how you switch between them

[Relevant thread](http://forum.caravelgames.com/viewtopic.php?TopicID=31280)

### CODE REVIEW

**Comment by mrimer** [comment]
Needs rebasing

File: DRODLib/DbRooms.cpp [hunk 1]
```diff
@@ -2924,30 +2925,18 @@ bool CDbRoom::DoesSquareContainTeleportationObstacle(
...
-		pMonster = pMonster->pNext; //start looking after this monster
-		if (!pMonster)
-			pMonster = this->pFirstMonster;
-	}
-
-	while (pMonster)
-	{
-		if (pMonster->wType == M_CLONE)
-			break; //found next clone
+	CClone* pFirstClone = DYN_CAST(CClone*, CMonster*, this->GetMonsterOfType(M_CLONE));
```
Comment (mrimer) [inline DRODLib/DbRooms.cpp:2928]: Could remove "this->" here, depending on your style.

**Review by mrimer — COMMENTED** [review]
Awesome code!

I made some minor stylistic comments.

Besides the rebase, I think this is ready to go.

**Comment by EvidentlyCube** [comment]
Thanks Mike, updated the PR

**Comment by EvidentlyCube** [comment]
Also added tests that I forgot to include in the commit in this and previous commits.

**Review by mrimer — APPROVED** [review]

**Pull request merged**